logtail/logtail.go

```diff
@@ -84,6 +98,12 @@ func NewLogger(cfg Config, logf tslogger.Logf) *Logger {
 		}
 		cfg.Buffer = NewMemoryBuffer(pendingSize)
 	}
+	var procID uint32
+	if cfg.IncludeProcId {
+		keyBytes := make([]byte, 4)
+		rand.Read(keyBytes)
+		procID = binary.LittleEndian.Uint32(keyBytes)
```

Comment:
In the 1 in 4 billion case where this is 0, we should add 1 since we specially treat 0 elsewhere as not being specified.